website/docs/reference/global-configs/behavior-changes.md

```diff
@@ -225,3 +228,47 @@ Once the metric is updated, it will work as expected:
         window: 7 days
 
 ```
+
+### Macro argument validation
+
+dbt supports optional validation for macro arguments using the `validate_macro_args` flag. By default, the `validate_macro_args` flag is set to `False`, which means that dbt won't validate the names or types of documented macro arguments.
+
+In the past, dbt didn't enforce a standard vocabulary for the [`type`](/reference/resource-properties/argument-type) field on macro arguments in YAML. Because of this, the `type` field was used for documentation only, and dbt didn't check that:
+- the argument names matched those in your macro
+- the argument types were valid or consistent with the macro's Jinja definition
+
+Here's an example of a documented macro:
+<File name='macros/filename.yml'>
+
+```yaml
+version: 2
+
+macros:
+  - name: <macro name>
+    arguments:
+      - name: <arg name>
+        type: <string>
+```
+</File>
+
+When you set the `validate_macro_args` flag to `True`, dbt will:
+- Check that all argument names in your YAML match those in the macro definition
+- Raise warnings if the names or types don't match
+- Validate that the `types` values follow the supported format explained in the next section
+- If no arguments are documented in the YAML, dbt will infer them from the macro and include them in the [`manifest.json` file](/reference/artifacts/manifest-json)
```

Comment:
@peterallenwebb is this last bullet a part of the behavior change flag? or does this happen regardless now?

Reply:
Brought this into Slack and will follow up with the answer there!